aeron-driver/src/main/c/aeron_publication_image.c

```diff
@@ -272,16 +272,31 @@ int aeron_publication_image_close(aeron_counters_manager_t *counters_manager, ae
         aeron_free(subscribable->array);
         aeron_free(image->connections.array);
 
-        image->raw_log_close_func(&image->mapped_raw_log, image->log_file_name);
         image->congestion_control->fini(image->congestion_control);
-        aeron_free(image->log_file_name);
     }
 
-    aeron_free(image);
-
     return 0;
 }
 
+bool aeron_publication_image_free(aeron_publication_image_t *image)
+{
+    if (NULL == image)
+    {
+        return true;
+    }
+
+    if (!image->raw_log_free_func(&image->mapped_raw_log, image->log_file_name))
+    {
+        return false;
+    }
+
+    aeron_free(image->log_file_name);
+    aeron_free(image);
+    return true;
+}
+
+extern bool aeron_publication_image_free_voidp(void *image);
```

Comment:
Could we avoid having an extra wrapper function by using `void *` in the actual function definition?
```C
bool aeron_publication_image_free(void *image)
```

Reply:
My preference would be for the typesafe version the voidp wrapper only being a necessity for the internals of the conductor.  Perhaps it would be better there.